pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -165,7 +169,7 @@ func (c *Controller) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 		deployInfrastructure = g.Add(flow.Task{
 			Name:         "Deploying Shoot infrastructure",
 			Fn:           flow.TaskFn(botanist.DeployInfrastructure).RetryUntilTimeout(defaultInterval, defaultTimeout),
-			Dependencies: flow.NewTaskIDs(deploySecrets, deployCloudProviderSecret, deployReferencedResources),
+			Dependencies: flow.NewTaskIDs(deploySecrets, deployCloudProviderSecret, deployReferencedResources, ensureShootExtensionStatus),
```

Comment:
Why is "only" this task depending on `ensureShootExtensionStatus`? I suggest to remove it here because, consequently, you would have to add it to all other tasks creating extension resources as well.

Reply:
> I suggest to remove it here  

where would you put it instead?